firebase-firestore/src/main/java/com/google/firebase/firestore/model/TargetIndexMatcher.java

```diff
@@ -186,6 +186,34 @@ public boolean servedByIndex(FieldIndex index) {
     return true;
   }
 
+  public FieldIndex BuildTargetIndex() {
```

Comment:
Even with a prototype, this code should be tested. CSI is complicated enough to warrant some tests.

Specifically, I want to see:

A list of query/targets feed to the matcher and get `FieldIndex` from buildTargetIndex().

All these generated indexes should all have `servedByIndex()` return true..they should all be a full match too.